File: api/src/main/java/com/hirrao/javaee/service/DrugAlertService.java
package com.hirrao.javaee.service;

import org.apache.ibatis.annotations.Delete;
import org.apache.ibatis.annotations.Update;
import org.springframework.stereotype.Service;

import java.sql.Date;
import java.sql.Time;

@Service
public interface DrugAlertService {
    void insertDrugAlert(long alertId, long uid, long drugId, Time alertTime, Date eatTime);

    void updateDrugAlertIsActiveById(long alertId,int isActive);

    void updateDrugAlertEatTimeById(long alertId,Date eatTime);

    void deleteDrugAlertById(long alertId,long uid,long drugId);
}

File: api/src/main/kotlin/com/hirrao/javaee/service/DrugAlertService.kt
package com.hirrao.javaee.service

import org.springframework.stereotype.Service
import java.time.LocalDate
import java.time.LocalTime

@Service
interface DrugAlertService {
    fun insertDrugAlert(alertId: Long, uid: Int, drugId: Long, alertTime: LocalTime?, eatTime: LocalDate?)

    fun updateDrugAlertIsActiveById(alertId: Long, isActive: Int)

    fun updateDrugAlertEatTimeById(alertId: Long, eatTime: LocalDate?)

    fun deleteDrugAlertById(alertId: Long, uid: Int, drugId: Long)
}
